### PR TITLE
CODEOWNERS: Add owners for core packages

### PR DESCRIPTION
**Pull request by christopherco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

The change adds codeowners for core packages, specifically packages for bootloaders, kernel, dracut, systemd, debug tools, openssl, package managers, container runtimes, virtualization, and provisioning. Aim is to have dedicated reviews from these codeowners to improve the quality of these packages.

###### Change Log  

- Add codeowners to tie cbl-mariner teams to specific bootloaders, kernel, dracut, systemd, debug tools, openssl, package managers, container runtimes, virtualization, and provisioning

###### Does this affect the toolchain?  

**NO**

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -7,6 +7,57 @@
...
+/SPECS/grub2/* @microsoft/cbl-mariner-bootloader
+/SPECS/grubby/* @microsoft/cbl-mariner-bootloader
+/SPECS/shim/* @microsoft/cbl-mariner-bootloader
+/SPECS/shim-unsigned/* @microsoft/cbl-mariner-bootloader
+/SPECS/shim-unsigned-x64/* @microsoft/cbl-mariner-bootloader
+/SPECS/shim-unsigned-aarch64/* @microsoft/cbl-mariner-bootloader
+/SPECS-SIGNED/grub2-efi-binary-signed/* @microsoft/cbl-mariner-bootloader
+
+/SPECS/dracut/* @microsoft/cbl-mariner-dracut
+/SPECS/initramfs/* @microsoft/cbl-mariner-dracut
```
Comment (dmcilvaney) [inline .github/CODEOWNERS:32]: Do we want to also add some of the [watched packages](https://github.com/microsoft/CBL-Mariner/blob/2.0/SPECS/initramfs/initramfs.spec#L22) for the initramfs?
```spec
%define watched_pkgs e2fsprogs, systemd, kpartx, device-mapper-multipath, verity-read-only-root, dracut-fips, dracut-megaraid
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -7,6 +7,57 @@
...
 /SPECS/core-packages/* @microsoft/cbl-mariner-admins
 
+# Modification to specific packages go to specific teams
+/SPECS/installkernel/* @microsoft/cbl-mariner-kernel
+/SPECS/kernel/* @microsoft/cbl-mariner-kernel
+/SPECS/kernel-azure/* @microsoft/cbl-mariner-kernel
+/SPECS/kernel-hci/* @microsoft/cbl-mariner-kernel
+/SPECS/kernel-headers/* @microsoft/cbl-mariner-kernel
+/SPECS/kernel-mshv/* @microsoft/cbl-mariner-kernel
+/SPECS/kernel-uvm/* @microsoft/cbl-mariner-kernel
```
Comment (rlmenge) [inline .github/CODEOWNERS:17]: Should we also include `kernel-mstflint` ?

**Review by rlmenge — APPROVED** [review]
Approved for CBL-Mariner-kernel :)

**Review by Camelron — APPROVED** [review]

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -7,6 +7,57 @@
 # Modification to what is considered "core packages" require admin approval.
 /SPECS/core-packages/* @microsoft/cbl-mariner-admins
```
[the remark this answers is not in the document]
Comment (hbeberman) [reply .github/CODEOWNERS:8]: You can look for teams in the Microsoft org: https://github.com/orgs/microsoft/teams?query=cbl-mariner-

[the remark this answers is not in the document]
Comment (hbeberman) [reply .github/CODEOWNERS:8]: You can also look for every team you're in with this query: https://github.com/orgs/microsoft/teams?query=+members%3Ame

**Review by hbeberman — APPROVED** [review]

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -7,6 +7,57 @@
...
+/SPECS/dnf-plugins-core/* @microsoft/cbl-mariner-package-managers
+/SPECS/rpm/* @microsoft/cbl-mariner-package-managers
+/SPECS/tdnf/* @microsoft/cbl-mariner-package-managers
+
+/SPECS/moby-buildx/* @microsoft/cbl-mariner-container-runtime
+/SPECS/moby-cli/* @microsoft/cbl-mariner-container-runtime
+/SPECS/moby-containerd/* @microsoft/cbl-mariner-container-runtime
+/SPECS/moby-engine/* @microsoft/cbl-mariner-container-runtime
+/SPECS/moby-runc/* @microsoft/cbl-mariner-container-runtime
+/SPECS/kata-containers/* @microsoft/cbl-mariner-container-runtime
```
Comment (dallasd1) [inline .github/CODEOWNERS:54]: We may want to have kata-containers-cc and moby-containerd-cc here as well

**Review by dallasd1 — APPROVED** [review]

**Review by sameluch — APPROVED** [review]

**Review by PawelWMS — APPROVED** [review]
I wasn't able to verify all groups, but the paths for the groups I'm a member of look good to me.

File: .github/CODEOWNERS (hunk 3 above)
[the remark this answers is not in the document]
Comment (christopherco) [reply .github/CODEOWNERS:8]: All of the above works. Also, each of the maintainer teams are GitHub child teams of the CBL-Mariner. team.https://github.com/orgs/microsoft/teams/cbl-mariner-devs/teams

File: .github/CODEOWNERS (hunk 4 above)
Comment (christopherco) [reply .github/CODEOWNERS:64]: Added

File: .github/CODEOWNERS (hunk 1 above)
Comment (christopherco) [reply .github/CODEOWNERS:40]: Added verity-read-only-root packages

**Review by dmcilvaney — APPROVED** [review]

**Pull request merged**